### PR TITLE
doc: update hash to curve rfc ref and impl comment

### LINKED ISSUE #477: bug: possibly incorrect `DST_prime` in `ExpandMsgXmd`
Unless I am missing something, it look like instead of `DST_prime = I2OSP(len(DST), 1) ∥ DST`, we do `DST_prime = DST || I2OSP(len(DST), 1) ` in:
- https://github.com/Consensys/gnark-crypto/blob/eb75782795d2d3ce6172797615f8eada77a6a3dd/field/hash/hashutils.go#L37-L40
- https://github.com/Consensys/gnark-crypto/blob/eb75782795d2d3ce6172797615f8eada77a6a3dd/field/hash/hashutils.go#L53-L56
- https://github.com/Consensys/gnark-crypto/blob/eb75782795d2d3ce6172797615f8eada77a6a3dd/field/hash/hashutils.go#L77-L80

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
fixes #477

### CODE REVIEW

File: field/hash/hashutils.go [hunk 1]
```diff
@@ -6,8 +6,8 @@ import (
 )
 
 // ExpandMsgXmd expands msg to a slice of lenInBytes bytes.
-// https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-06#section-5
-// https://tools.ietf.org/html/rfc8017#section-4.1 (I2OSP/O2ISP)
+// https://tools.ietf.org/doc/html/rfc9380#name-expand_message_xmd
```
Comment (ivokub) [inline field/hash/hashutils.go:9]: The links redirect to IETF home page. Maybe use https://www.rfc-editor.org/rfc/rfc9380.html instead?

**Review by ivokub — CHANGES_REQUESTED** [review]

File: field/hash/hashutils.go (hunk 1 above)
Comment (ivokub) [reply field/hash/hashutils.go:9]: Or use https://datatracker.ietf.org/doc/html/rfc9380#name-expand_message_xmd

Comment (hussein-aitlahcen) [reply field/hash/hashutils.go:9]: sure

**Comment by hussein-aitlahcen** [comment]
@ivokub I updated the changes

**Review by ivokub — APPROVED** [review]
Perfect! Thanks for the contribution!

**Pull request merged**